src/messaging/ExchangeContext.cpp

```diff
@@ -266,6 +267,27 @@ CHIP_ERROR ExchangeContext::SendMessage(Protocols::Id protocolId, uint8_t msgTyp
     }
 }
 
+void ExchangeContext::SendStatusReport(const Protocols::SecureChannel::StatusReport & statusReport)
+{
+    ChipLogDetail(ExchangeManager, "Sending status report. Protocol code %u, exchange %u", statusReport.GetProtocolCode(),
+                  GetExchangeId());
+
+    auto handle = System::PacketBufferHandle::New(statusReport.Size());
+    VerifyOrReturn(!handle.IsNull(), ChipLogError(SecureChannel, "Failed to allocate status report message"));
+    Encoding::LittleEndian::PacketBufferWriter bbuf(std::move(handle));
+
+    statusReport.WriteToBuffer(bbuf);
+
+    System::PacketBufferHandle msg = bbuf.Finalize();
+    VerifyOrReturn(!msg.IsNull(), ChipLogError(SecureChannel, "Failed to finalize status report message"));
+
+    CHIP_ERROR err = SendMessage(Protocols::SecureChannel::MsgType::StatusReport, std::move(msg));
```

Comment:
So how does the caller know whether SendMessage succeeded?  They may need to know that to do proper lifetime management on the exchange.

This method should be returning CHIP_ERROR, not void.  Consumers that don't care can ignore the return value.